cpp/docs/TRANSITIONGUIDE.md

```diff
@@ -125,84 +177,199 @@ auto return_two_outputs(void){
 CUDA_TRY( cudaMemcpy(&dst, &src, num_bytes) );
 ```
 
-## Testing
 
-### ```column_wrapper```
+## Compile-Time Conditions
 
-The `column_wrapper<T>` class template is defined in `cudf/cpp/tests/utilities/legacy/column_wrapper.cuh` to simplify the creation and management of `gdf_column`s for the purposes of unit testing. 
+Some conditions can be verified at compile time. These should be done using `static_assert`. For example,
 
-`column_wrapper<T>` provides a number of constructors that allow easily constructing a `gdf_column` with the appropriate `gdf_dtype` enum set based on mapping `T` to an enum, e.g., `column_wrapper<int>` will correspond to a `gdf_column` whose `gdf_dtype` is set to `GDF_INT32`.
+```c++
+template <typename T>
+void trivial_types_only(T t){
+   static_assert(std::is_trivial<T>::value, "This function requires a trivial type.");
+...
+}
+```
 
-The simplest constructor creates an unitilized `gdf_column` of a specified type with a specified size:
 
-```
-cudf::test::column_wrapper<T>  col(size);
-```
+# Testing
 
-You can also construct a `gdf_column` that uses a `std::vector` to initialize the `data` and `valid` bitmask of the `gdf_column`.
+TBD
 
-```
- std::vector<T> values(size);
 
- std::vector<gdf_valid_type> expected_bitmask(gdf_valid_allocation_size(size), 0xFF);
 
- cudf::test::column_wrapper<T> const col(values, bitmask);
-```
+# Porting Guide
 
-Another constructor allows passing in an initializer function that accepts a row index that will be invoked for every index `[0, size)` in the column:
+This section provides the high-level steps necessary to port an existing `libcudf` API into a `libcudf++` function. 
 
+For example, given the example function `old_function`:
 ```
-  // This creates a gdf_column with data elements {0, 1, ..., size-1} with a valid bitmask
-  // that indicates all of the values are non-null
-  cudf::test::column_wrapper<T> col(size, 
-      [](auto row) { return row; },
-      [](auto row) { return true; });
-```
-
-To access the underlying `gdf_column` for passing into a libcudf function, the `column_wrapper::get` function can be used to provide a pointer to the underlying `gdf_column`.
+// cpp/include/cudf/utilities/old.hpp
 
+namespace cudf{
+/**
+* @brief This is an example of an old API.
+*
+* @param input Immutable input column
+* @param in_out Mutable column modified in place
+* @param input_table Immutable input table
+* @param in_out_table Mutable table modified in place
+* @return gdf_column Newly allocated output column
+**/
+
+gdf_column  old_function(gdf_column const& input, 
+                         gdf_column* in_out,
+                         cudf::table const& input_table,
+                         cudf::table* in_out_table);
+} // namespace cudf
 ```
-column_wrapper<T> col(size);
-gdf_column* gdf_col = col.get();
-some_libcudf_function(gdf_col...);
+
+ 1. Move old header and source/tests into `legacy/` directories
+	 - `mkdir -p cudf/cpp/include/cudf/utilities/legacy`
```

Comment:
The header file path needs to be updated to have legacy in other source files, we can just add the legacy path in the new header file to reduce the burden.

Reply:
That won't work because the include paths are usually explicit, like `#include <cudf/copying.hpp>`. 

These need to be updated to `#include <cudf/legacy/copying.hpp>`

Reply:
And remember to update `#ifndef #defines` for the header files.

Reply:
You can add #include <cudf/legacy/copying.hpp> into cudf/copying.hpp, you don't have to update all the files.

Reply:
> You can add #include <cudf/legacy/copying.hpp> into cudf/copying.hpp, you don't have to update all the files.

While this would work, I'd like to keep the new and legacy code as separate as possible.